### PR TITLE
Tweaked one-of diffing

### PR DESCRIPTION
**Pull request by liujed**
Lifted additions and removals at the level of one-of variants to changes at the level of the enclosing one-of.

### CODE REVIEW

File: visitors/http_rest_diff/spec_diff_visitor.go [hunk 1]
```diff
@@ -210,9 +213,39 @@ func (*DefaultSpecDiffVisitorImpl) EnterOneOfs(self interface{}, ctx http_rest.S
...
+			if IsSameData(leftOption, rightOption) {
+				delete(rightOptions, rightKey)
+				continue OUTER
+			}
+		}
+
+		// No match found for leftOption.
+		return true
+	}
+	return false
```
Comment (mgritter) [inline visitors/http_rest_diff/spec_diff_visitor.go:246]: I don't think this is right.

Consider leftOption = {A} and rightOption = {A, B}

I think you meant 
```
return len( rightOption ) == 0
```

Comment (liujed) [reply visitors/http_rest_diff/spec_diff_visitor.go:246]: Isn't this covered by the length check above the loop?

Comment (mgritter) [reply visitors/http_rest_diff/spec_diff_visitor.go:246]: Hm, OK, you're right, each loop should decrease the length of rightOptions by one or terminate.

**Review by mgritter — APPROVED** [review]

File: visitors/http_rest_diff/spec_diff_visitor.go (hunk 1 above)
Comment (liujed) [reply visitors/http_rest_diff/spec_diff_visitor.go:246]: Let's port to F* real quick and verify this! 😄

**Comment by mgritter** [comment]
OK, is the problem that the visitor previously visited each option in an unpredictable order, leading to false positives, but now it removes all those false positives -- but, the add and remove will still look the same for oneOfs that actually have differences?

**Comment by liujed** [comment]
Answered here: https://akitasoftware.slack.com/archives/CSL17U42G/p1654565899003119.

**Pull request merged**